include/ibamr/CarmanKozenyDragForce.h

```diff
@@ -0,0 +1,165 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (c) 2019 - 2021 by the IBAMR developers
+// All rights reserved.
+//
+// This file is part of IBAMR.
+//
+// IBAMR is free software and is distributed under the 3-clause BSD
+// license. The full text of the license can be found in the file
+// COPYRIGHT at the top level directory of IBAMR.
+//
+// ---------------------------------------------------------------------
+
+/////////////////////////////// INCLUDE GUARD ////////////////////////////////
+
+#ifndef included_CarmanKozenyDragForce
+#define included_CarmanKozenyDragForce
+
+/////////////////////////////// INCLUDES /////////////////////////////////////
+#include "ibamr/BrinkmanPenalizationStrategy.h"
+
+/////////////////////////////// NAMESPACE ////////////////////////////////////
+
+namespace IBAMR
+{
+class INSVCStaggeredHierarchyIntegrator;
+class AdvDiffHierarchyIntegrator;
+} // namespace IBAMR
+namespace SAMRAI
+{
+namespace pdat
+{
+template <int DIM, class TYPE>
+class CellVariable;
+} // namespace pdat
+} // namespace SAMRAI
+
+/////////////////////////////// CLASS DEFINITION /////////////////////////////
+
+namespace IBAMR
+{
+/*!
+ * \brief CarmanKozenyDragForce provides an implementation of Carman-Kozeny drag force
+ * to impose zero velocity inside the solid \f$ \bm{u}=\bm{u}_b = 0\f$.
+ *
+ * The penalization force is taken to be \f$ A_d(\bm{u}_b - \bm{u}^{n+1}) \f$. The class
+ * computes the coefficient \f$A_d \f$ of the fluid velocity \f$ \bm{u}^{n+1} \f$ for the
+ * variable-coefficient INS solvers of INSVCStaggeredHierarchyIntegrator. This is done in
+ * the CarmanKozenyDragForce::demarcateBrinkmanZone method. Here \f$ A_d =
+ * C_d\frac{\alpha_S}{(1-alpha_S)^3+e_d}\f$, \f$ \alpha_S \f$ is the volume fraction of
+ * the solid, \f$ C_d\f$ and  \f$ e_d\f$ are the model parameters. The rigid body velocity
+ * \f$\bm{u}_b\f$ is taken to be zero. The penalty
+ * parameter C_d is taken to be \f$C_d = ( \rho / \Delta t + \mu / h^2)\f$. The user can choose
+ * the density or the inertia scale or both for \f$C_d\f$ through input.
+ */
+class CarmanKozenyDragForce : public BrinkmanPenalizationStrategy
+{
+public:
+    /*
+     * \brief Constructor of the class.
+     */
+    CarmanKozenyDragForce(std::string object_name,
+                          SAMRAI::tbox::Pointer<SAMRAI::pdat::CellVariable<NDIM, double> > H_var,
+                          SAMRAI::tbox::Pointer<SAMRAI::pdat::CellVariable<NDIM, double> > lf_var,
+                          SAMRAI::tbox::Pointer<IBAMR::AdvDiffHierarchyIntegrator> adv_diff_solver,
+                          SAMRAI::tbox::Pointer<IBAMR::INSVCStaggeredHierarchyIntegrator> fluid_solver,
+                          SAMRAI::tbox::Pointer<SAMRAI::tbox::Database> input_db = nullptr,
```

Comment:
```suggestion
                          SAMRAI::tbox::Pointer<SAMRAI::tbox::Database> input_db,
```
input databases should not be optional